lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,127 @@
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain subnets for section_id %s from phpIPAM."), section_id.to_s)
+    end
+
+    # Gets the subnet in phpIPAM, taking subnet in cidr format as parameter(e.g. 100.20.20.0/24)
+    # Returns: JSON object with "data" key if exists, otherwise JSON object with "message" key
+    #          containing an error message.
+    #   Example(exists):
+    #     {"code":200,"success":true,"data":[{"id":"12","subnet":"100.30.30.0","mask":"24","sectionId":"5",
+    #      "description":"New Subnet for Demo","linked_subnet":null,"firewallAddressObject":null,"vrfId":"0",
+    #      "masterSubnetId":"0","allowRequests":"0","vlanId":"0","showName":"0","device":"0","permissions":"[]",
+    #      "pingSubnet":"0","discoverSubnet":"0","DNSrecursive":"0","DNSrecords":"0","nameserverId":"0",
+    #      "scanAgent":"0","isFolder":"0","isFull":"0","tag":"2","threshold":"0","location":"0","editDate":null,
+    #      "lastScan":null,"lastDiscovery":null}],"time":0.009}
+    #   Example(not exists): {"code":200,"success":0,"message":"No subnets found","time":0.01}
+    def get_subnet(subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      get("/get_subnet?cidr=#{cidr}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain subnet %s from phpIPAM."), cidr)
+    end
+
+    # Checks whether an IP address has already been taken in phpIPAM
+    # Returns: JSON object with 'exists' field being either true or false
+    #   Example: {"ip":"100.20.20.18","exists":false}
+    def ip_exists(ip, subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      get("/ip_exists?cidr=#{cidr}&ip=#{ip}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain IP address for subnet_id %s from phpIPAM."), subnet['id'].to_s)
+    end
+
+    # Deletes IP address from a given subnet
+    # Returns: JSON object
+    #   Example(success): {"code"=>200, "success"=>true, "message"=>"Address deleted", "time"=>0.017}
+    #   Example(error):   {"code"=>404, "success"=>0, "message"=>"Address does not exist", "time"=>0.008}
+    def delete_ip_from_subnet(ip, subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      parse(post({:ip => ip, :cidr => cidr}, "/delete_ip_from_subnet"))
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to delete IP %s from the subnet %s in phpIPAM."), ip, cidr)
+    end
+
+    private
+
+    # TODO: Use ProxyAPI::Resource.parse instead for GET(for consistency), once it has been
+    #       refactored to handle error messages better
+    def get(path, body = nil)
+      uri = URI(@url + path)
+      uri.query = URI.encode_www_form(body) if body
+      request = Net::HTTP::Get.new(uri)
+      response = Net::HTTP.start(uri.hostname, uri.port) {|http| http.request(request)}
+      JSON.parse(response.body)
+    end
+  end
```

Comment:
If you can improve our current method, do it but take special care - tests must pass, this is indeed not the best part of our codebase :-)

Reply:
Perhaps I will take a look, but will be in a different PR